scripts/generate_certs.py

```diff
@@ -0,0 +1,156 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import sys
+import os
+import re
+import subprocess
+import tempfile
+
+class UpdaterCerts:
+    """ Updater for certs.c"""
+
+    def __init__(self):
+        """Instantiate the updater for certs.c
+
+        certs: path to certs.c
+        """
+        self.certs = "tests/src/certs.c"
+
+    @staticmethod
+    def type_var_bin(name, filename):
+        """ Generates code snipper to be added
+        name: name of variable
+        filename: where value of variable stored
+        """
+        output = "const unsigned char " + name + "[] = {\n"
+        data = subprocess.check_output(["xxd", "-i", filename]).decode("utf-8")
+        output += '   ' + '   '.join(data.splitlines(keepends=True)[1:-2])
+        output += "};\n"
+        return output
+
+    @staticmethod
+    def type_var_string(name, filename):
+        """ Generates code snipper to be added
+        name: name of variable
+        filename: where value of variable stored
+        """
+        output = "const char " + name + "[] =\n"
+        with open(filename) as f:
+            for line in f.read().splitlines():
+                output += "    \"" + line + "\\r\\n\"\n"
+            output = output[:-1] + ";\n"
+        return output
+
+    @staticmethod
+    def type_macro_bin(name, filename):
+        """ Generates code snipper to be added
+        name: name of macro
+        filename: where value of macro stored
+        """
+        output = "#define " + name + " {\n"
+        data = subprocess.check_output(["xxd", "-i", filename]).decode("utf-8")
+        output += '  ' + '  '.join(data.splitlines(keepends=True)[1:-2])
+        output = "".join([(line + (77-len(line))*" " + "\\\n") for line in output.splitlines()])
+        output += "}\n"
+        return output
+
+    @staticmethod
+    def type_macro_string(name, filename):
+        """ Generates code snipper to be added
+        name: name of macro
+        filename: where value of macro stored
+        """
+        output = "#define " + name + "\n"
+        with open(filename) as f:
+            for line in f.read().splitlines()[:-1]:
+                output += "    \"" + line + "\\r\\n\"\n"
+            output = output[:-1] + "\n"
+        output = "".join([(line + (75-len(line))*" " + "\\\n") for line in output.splitlines()])
+        with open(filename) as f:
+            output += "    \"" + f.read().splitlines()[-1] + "\\r\\n\"\n"
+        return output
+
+    # Extract key or certificate from file.
+    def extract_key(self, enc, typ, name, filename):
+        """ Depending on encoding and type it calls appriopriate
+        function
+
+        enc: encoding
+        typ: type
+        name: name of variable/macro
+        filename: where data is
+        """
+        if enc not in ("string", "binary"):
+            print("choose either string encoding or binary encoding")
+            sys.exit(1)
```

Comment:
Would potentially be better to raise a custom exception here, caught by a try/except block in Update(), rather than just sys.exit(). Certainly we should be reporting what the type passed in was.

Reply:
added new custom exceptions. Catches it in main and sys exit from run_main() function